Makefile

```diff
@@ -14,6 +14,15 @@ docker-test:
 	docker build . -t otter-test -f test-Dockerfile --cache-from ucbdsinfra/otter-grader:latest
 	rm test-Dockerfile
 
+docker-ci-test:
+	cp -r Dockerfile test-Dockerfile
+	printf "\nADD . /home/otter-grader\nRUN pip install /home/otter-grader" >> test-Dockerfile
+	printf "\nRUN cd /tmp && curl -sSL -O https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}.tgz && tar zxf docker-${DOCKER_VERSION}.tgz && mv ./docker/docker /usr/local/bin && chmod +x /usr/local/bin/docker && rm -rf /tmp/*" >> test-Dockerfile
+	printf "\nCOPY --from=docker/buildx-bin /buildx /usr/libexec/docker/cli-plugins/docker-buildx" >> test-Dockerfile
+	printf "\nENV PYTHONUNBUFFERED 1" >> test-Dockerfile
+	docker buildx build . --load -t otter-test -f test-Dockerfile --cache-from=type=gha --cache-to=type=gha,mode=max
```

Comment:
Can we out the first 5 lines of this target into a separate one that `docker-test` and `docker-ci-test` rely on? e.g.

```
docker-test-precursor:
	# those lines

docker-test: docker-test-precursor
	# etc.

docker-ci-test: docker-test-precursor
	# etc.
```